src/chart/draw/consolidateData/transformData.js

```diff
@@ -354,16 +186,33 @@ export default function transformData(raw, mark) {
           ? set(filtered.map(m => m[config.x.column])).values()
           : config.x.type === 'ordinal'
             ? set(raw.map(m => m[config.x.column])).values()
-            : config.x_from0 ? [0, max(pre_x_dom)] : pre_x_dom;
+            : pre_x_dom;
 
     let y_dom = config.y_dom
         ? config.y_dom
         : config.y.type === 'ordinal' && config.y.behavior === 'flex'
           ? set(filtered.map(m => m[config.y.column])).values()
           : config.y.type === 'ordinal'
             ? set(raw.map(m => m[config.y.column])).values()
-            : config.y_from0 ? [0, max(pre_y_dom)] : pre_y_dom;
-
+            : pre_y_dom;
+
+    //set lower limit of linear domain to 0 when other axis is ordinal, provided no values are negative
```

Comment:
I think this does it. Would probably be good to explain this behavior for "flex bar charts" in the wiki, and note that users can still explicitly set non-zero domains on bar charts by using `behavior` other than flex.